### PR TITLE
[#101692394] Email validation for supplier declaration

### PR DESCRIPTION
**Pull request by TheDoubleK**
See: https://www.pivotaltracker.com/story/show/101692394

This uses the regex from our User JSON schema to check for valid email addresses.

Content has already been updated with messaging.

### CODE REVIEW

**Comment by TheDoubleK** [comment]
Oh hang on...

**Comment by TheDoubleK** [comment]
Better now.

File: app/main/helpers/frameworks.py [hunk 1]
```diff
@@ -108,7 +108,10 @@ def get_formatting_errors(answers):
     errors_map = {}
     if not re.match(r'^(\S{9}|\S{12})$', answers.get('SQ1-1h', '')):
         errors_map['SQ1-1h'] = 'invalid_format'
-
+    if not re.match(r'^[^@^\s]+@[^@^\.^\s]+(\.[^@^\.^\s]+)+$', answers.get('SQ1-1o', '')):
```
Comment (allait) [inline app/main/helpers/frameworks.py]: Looking at the code it's not immediately obvious that this is email validation — I'd struggle to find this part if I were searching. I think saving the regex pattern to a named constant could help here a bit.

**Comment by TheDoubleK** [comment]
Good point - fixed up now.

**Comment by robyoung** [comment]
:+1:

**Pull request merged**